navpy/core/navpy.py

```diff
@@ -903,22 +903,25 @@ def ecef2lla(ecef, latlon_unit='deg'):
     alt : {(N,)} array like altitude in meters
     """
     ecef,N = _input_check_Nx3(ecef)
-    if(N>1):
-        x = ecef[:,0]; y = ecef[:,1]; z = ecef[:,2]
-    else:
-        x = ecef[0]; y = ecef[1]; z = ecef[2]
-
+    ecef = ecef.reshape(N,3)
```

Comment:
Use 2D array here to handle possible multiple ECEF coordinates. Will use indexing to separate latitudes near the poles and the ones further away from it.